lightgbm/src/main/scala/com/microsoft/azure/synapse/ml/lightgbm/params/BaseTrainParams.scala

```diff
@@ -0,0 +1,259 @@
+  {
+    sb.appendParamValueIfNotThere("is_pre_partition", Option("True"))
+      .appendParamValueIfNotThere("boosting_type", Option(boostingType))
+      .appendParamValueIfNotThere("tree_learner", Option(parallelism))
+      .appendParamValueIfNotThere("top_k", topK)
+      .appendParamValueIfNotThere("num_leaves", numLeaves)
+      .appendParamValueIfNotThere("max_bin", maxBin)
+      .appendParamValueIfNotThere("min_data_per_bin", minDataPerBin)
+      .appendParamValueIfNotThere("bagging_fraction", baggingFraction)
+      .appendParamValueIfNotThere("pos_bagging_fraction", posBaggingFraction)
+      .appendParamValueIfNotThere("neg_bagging_fraction", negBaggingFraction)
+      .appendParamValueIfNotThere("bagging_freq", baggingFreq)
+      .appendParamValueIfNotThere("feature_fraction", featureFraction)
+      .appendParamValueIfNotThere("max_depth", maxDepth)
+      .appendParamValueIfNotThere("min_sum_hessian_in_leaf", minSumHessianInLeaf)
+      .appendParamValueIfNotThere("lambda_l1", lambdaL1)
+      .appendParamValueIfNotThere("lambda_l2", lambdaL2)
+      .appendParamValueIfNotThere("metric", metric)
+      .appendParamValueIfNotThere("min_gain_to_split", minGainToSplit)
+      .appendParamValueIfNotThere("max_delta_step", maxDeltaStep)
+      .appendParamValueIfNotThere("min_data_in_leaf", minDataInLeaf)
+      .appendParamValueIfNotThere("num_iterations", Option(numIterations))
+      .appendParamValueIfNotThere("learning_rate", Option(learningRate))
+      .appendParamValueIfNotThere("num_machines", Option(numMachines))
+      .appendParamValueIfNotThere("verbosity", Option(verbosity))
+      .appendParamValueIfNotThere("early_stopping_round", Option(earlyStoppingRound))
+      .appendParamValueIfNotThere("learning_rate", Option(learningRate))
+      .appendParamValueIfNotThere("min_data_per_bin", minDataPerBin)
+      .appendParamListIfNotThere("categorical_feature", categoricalFeatures)
+      .appendParamListIfNotThere("max_bin_by_feature", maxBinByFeature)
+      .appendParamValueIfNotThere("top_rate", topRate)
+      .appendParamValueIfNotThere("other_rate", otherRate)
+      .appendParamListIfNotThere("monotone_constraints", monotoneConstraints)
+      .appendParamValueIfNotThere("monotone_constraints_method", monotoneConstraintsMethod)
+      .appendParamValueIfNotThere("monotone_penalty", monotonePenalty)
+      .appendParamValueIfNotThere("learning_rate", Option(learningRate))
+  }
+}
+
+/** Defines the Dataset parameters passed to the LightGBM classifier.
+  *
+  * @param isEnableSparse Used to enable/disable sparse optimization.
+  * @param useMissing Set this to false to disable the special handle of missing value.
+  * @param zeroAsMissing Set to true to treat all zero as missing values
+  *                         (including the unshown values in LibSVM/sparse matrices)
+  *                      Set to false to use na for representing missing values.
+  */
+case class DatasetParams(isEnableSparse: Option[Boolean],
```

Comment:
This is awesome!  Can you add tests for all of these new parameters?  It's not always as simple as adding them here, and sometimes the distributed version of lightgbm does not support them (like forcedsplits_filename, which apparently only works in single node case https://github.com/Microsoft/LightGBM/blob/master/docs/Parameters.rst#forcedsplits_filename and I only found out the hard way after adding it everywhere in scala but then realizing it just didn't do anything for distributed case).

Reply:
I added some tests for the new sparse parameters.  I also did some code analysis of the LightGBM code, and I don't see any limitations of any of the new parameters.  When looking at the one you mentioned (forcedsplits_filename), it does indeed show in the validation code that you can't run it under the types of parallelism we use (data and voting).

I'm not sure I see the utility of keeping those tests.  They take time, and once we know it works it should work always (if they ever make a breaking change we would break anyways). Can I just remove them?  They weren't fancy, they just proved that LightGBM had the right value for the parameters and that the fit succeeded.

Reply:
"I'm not sure I see the utility of keeping those tests."
It's useful to make sure that:
1.) The parameters actually do work right now, and with future code changes they will still work
2.) With new native lightgbm jar version updates in the future they still work, and names haven't changed, etc
3.) They serve as a good example for others trying to understand how the parameters work and what effect they have by looking at the test cases
At least that is my view anyway.  Without tests we don't really know if they work for sure.